main.tf

```diff
@@ -80,3 +89,17 @@ resource "google_project_iam_member" "pubsub_sa_create_oidc_token" {
   member     = "serviceAccount:${local.google_pubsub_sa_email}"
   depends_on = [google_project_service.pubsub_api]
 }
+
+resource "google_project_iam_member" "pubsub_sa_acknowledge_for_dead_letter" {
+  # See https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/pubsub_subscription#dead_letter_policy
+  role       = "roles/pubsub.subscriber"
+  member     = "serviceAccount:${local.google_pubsub_sa_email}"
+  depends_on = [google_project_service.pubsub_api]
+}
+
+resource "google_project_iam_member" "pubsub_sa_publish_to_dead_letter" {
+  # see https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/pubsub_subscription#dead_letter_topic
```

Comment:
Same comment as above and start the sentence with capital S to keep it consistent across 😆